src/main/java/io/ebean/common/ModifyHolder.java

```diff
@@ -54,6 +56,11 @@ private boolean undoDeletion(E bean) {
   void modifyAddition(E bean) {
     if (bean != null) {
       touched = true;
+
+      if (bean instanceof EntityBean) {
```

Comment:
This is the place, where the question came up, why BeanCollection (and ModifyHolder) doesn't know anything about EntityBean: https://groups.google.com/forum/#!topic/ebean/wsiYTGYHpCg